pylint/checkers/dataclass_checker.py

```diff
@@ -0,0 +1,114 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Dataclass checkers for Python code."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from astroid import nodes
+from astroid.brain.brain_dataclasses import DATACLASS_MODULES
+
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import INFERENCE
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+def _is_dataclasses_module(node: nodes.NodeNG) -> bool:
+    """Utility function to check if node is from dataclasses_module."""
+    return isinstance(node, nodes.Module) and node.name in DATACLASS_MODULES
+
+
+def _check_name_or_attrname_eq_to(
+    node: nodes.Name | nodes.Attribute, check_with: str
+) -> Any:
```

Comment:
```suggestion
) -> bool:
```

Reply:
I can't remember why but this doesn't pass type checking

Reply:
Ah okay, I can cast the .name / .attrname attributes to string

Reply:
We should add an ignore then. This is due to `astroid` missing annotations